webpack/JobInvocationDetail/JobInvocationToolbarButtons.js

```diff
@@ -0,0 +1,272 @@
+  const queryParams = new URLSearchParams({
+    [`report_template_report[input_values][${templateInputId}][value]`]: jobId,
+  });
+  const [isActionOpen, setIsActionOpen] = useState(false);
+  const [reportTemplateJobId, setReportTemplateJobId] = useState(undefined);
+  const [templateInputId, setTemplateInputId] = useState(undefined);
+
+  const dispatch = useDispatch();
+  // eslint-disable-next-line no-shadow
+  const onActionToggle = isActionOpen => {
+    setIsActionOpen(isActionOpen);
+  };
+  const onActionFocus = () => {
+    const element = document.getElementById(
+      'toggle-split-button-action-primary'
+    );
+    element.focus();
+  };
+  const onActionSelect = () => {
+    setIsActionOpen(false);
+    onActionFocus();
+  };
+  const hasPermission = permissionRequired =>
+    permissionsStatus === APIStatus.RESOLVED
+      ? currentPermissions?.some(
+          permission => permission.name === permissionRequired
+        )
+      : false;
+
+  useEffect(() => {
+    dispatch(
+      get({
+        key: GET_REPORT_TEMPLATES,
+        url: '/api/report_templates',
+        handleSuccess: ({ data: { results } }) => {
+          setReportTemplateJobId(
+            results.find(result => result.name === 'Job - Invocation Report')
+              ?.id
+          );
+        },
+        handleError: () => {
+          setReportTemplateJobId(undefined);
+        },
+      })
+    );
+  }, [dispatch]);
+  useEffect(() => {
+    if (reportTemplateJobId !== undefined) {
+      dispatch(
+        get({
+          key: GET_REPORT_TEMPLATE_INPUTS,
+          url: `/api/templates/${reportTemplateJobId}/template_inputs`,
+          handleSuccess: ({ data: { results } }) => {
+            setTemplateInputId(
+              results.find(result => result.name === 'job_id')?.id
+            );
+          },
+          handleError: () => {
+            setTemplateInputId(undefined);
+          },
+        })
+      );
+    }
+  }, [dispatch, reportTemplateJobId]);
+
+  const recurrenceDropdownItems = recurrence
+    ? [
+        <DropdownSeparator ouiaId="dropdown-separator-1" key="separator-1" />,
+        <DropdownItem
+          ouiaId="change-enabled-recurring-dropdown-item"
+          onClick={() =>
+            dispatch(
+              enableRecurringLogic(recurrence?.id, recurringEnabled, jobId)
+            )
+          }
+          key="change-enabled-recurring"
+          component="button"
+          isDisabled={
+            recurrence?.id === undefined ||
+            recurrence?.state === 'cancelled' ||
+            !canEditRecurringLogic
+          }
+        >
+          {recurringEnabled ? __('Disable recurring') : __('Enable recurring')}
+        </DropdownItem>,
+        <DropdownItem
+          ouiaId="cancel-recurring-dropdown-item"
+          onClick={() => dispatch(cancelRecurringLogic(recurrence?.id, jobId))}
+          key="cancel-recurring"
+          component="button"
+          isDisabled={
```

Comment:
I think that if you cancel the recurring you can't cancel/enable/disable it anymore, so that's why I do it, but maybe I misunderstood your question.

Reply:
> Do you suggest not displaying the buttons completely if it's canceled? I leave them disabled so it's clear a job was in recurrence before and now it's not, but I can of course change it if you think otherwise.

yes thats what I meant, @adamruzicka should we let users know that the job was recurring?

Reply:
> should we let users know that the job was recurring?
> It would be nice to have it there somewhere

Maybe I could add it later on here:
![image](https://github.com/theforeman/foreman_remote_execution/assets/78563507/bca0456a-6729-43e4-94a2-e143b382ad3e)

Reply:
As a rule of a thumb, I would base the decision on whether the action is unavailable generally or locally.

I would hide a button if that button's action is generally not applicable, for example I would hide the "disable/enable recurring" button if the job is not recurring at all.

I would mark it as disabled if that button's action is locally not applicable, for example I would disable the "cancel" button on a job that already finished.

If I'm reading it correctly then the code adheres to this so I would leave it as is

Reply:
> If I'm reading it correctly then the code adheres to this so I would leave it as is

I agree, I already changed it, but I'll change it back.